Code/Framework/AzFramework/AzFramework/DocumentPropertyEditor/CvarAdapter.cpp

```diff
@@ -0,0 +1,198 @@
+                    contents.ArrayPushBack(Dom::Value(container.GetElement(i)));
+                }
+                builder.BeginPropertyEditor<NodeType>(AZStd::move(contents));
+                builder.OnEditorChanged(
+                    [this, functor](const Dom::Path& path, const Dom::Value& value)
+                    {
+                        // ConsoleCommandContainer holds string_views, so ensure we allocate our parameters here
+                        AZStd::fixed_vector<CVarFixedString, ElementCount> newValue;
+                        for (int i = 0; i < ElementCount; ++i)
+                        {
+                            newValue.push_back(ConsoleTypeHelpers::ValueToString(value[i].GetDouble()));
+                        }
+                        (*functor)(ConsoleCommandContainer(newValue));
+                        m_adapter->OnContentsChanged(path, value);
+                    });
+                builder.EndPropertyEditor();
+                return true;
+            }
+            return false;
+        }
+
+        bool TryBuildCheckBoxEditor(AdapterBuilder& builder, ConsoleFunctorBase* functor)
+        {
+            if (functor->GetTypeId() == azrtti_typeid<bool>())
+            {
+                bool value;
+                functor->GetValue(value);
+                builder.BeginPropertyEditor<Nodes::CheckBox>(Dom::Value(value));
+                builder.OnEditorChanged(
+                    [this, functor](const Dom::Path& path, const Dom::Value& value)
+                    {
+                        (*functor)({ ConsoleTypeHelpers::ValueToString(value.GetBool()) });
+                        m_adapter->OnContentsChanged(path, value);
+                    });
+                builder.EndPropertyEditor();
+                return true;
+            }
+            return false;
+        }
+
+        bool BuildEditorForCvar(AdapterBuilder& builder, ConsoleFunctorBase* functor)
+        {
+            // Check all possible type IDs for our supported types and try to create a property editor
+            return TryBuildNumericEditor<int8_t, int16_t, int32_t, int64_t, uint8_t, uint16_t, uint32_t, uint64_t, float, double>(
+                       builder, functor) ||
+                TryBuildStringEditor(builder, functor) || TryBuildMathVectorEditor<AZ::Vector2, 2, Nodes::Vector2>(builder, functor) ||
+                TryBuildMathVectorEditor<AZ::Vector3, 3, Nodes::Vector3>(builder, functor) ||
+                TryBuildMathVectorEditor<AZ::Vector4, 4, Nodes::Vector4>(builder, functor) ||
+                TryBuildMathVectorEditor<AZ::Quaternion, 4, Nodes::Quaternion>(builder, functor) ||
+                TryBuildMathVectorEditor<AZ::Color, 4, Nodes::Color>(builder, functor) || TryBuildCheckBoxEditor(builder, functor);
+        }
+
+        CvarAdapter* m_adapter;
+    };
+
+    CvarAdapter::CvarAdapter()
+        : m_impl(AZStd::make_unique<Impl>(this))
+    {
+    }
+
+    CvarAdapter::~CvarAdapter()
+    {
+    }
```

Comment:
can be set to default if no special actions are needed here?

Reply:
Little C++ gotcha - `= default;` will fail because we have a forward declared type (`Impl`) that won't have an accessible dtor necessarily available in the header file when the default constructor tries to call `~AZStd::unique_ptr<Impl>()` for the undefined type.